src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1255,8 +1255,13 @@ private ImmutableHashSet<DiagnosticAnalyzer> ComputeSuppressedAnalyzersForTree(S
                 var hasUnsuppressedDiagnostic = false;
                 foreach (var descriptor in descriptors)
                 {
-                    if (!options.TryGetDiagnosticValue(tree, descriptor.Id, out var configuredSeverity) ||
-                        configuredSeverity != ReportDiagnostic.Suppress)
+                    options.TryGetGlobalDiagnosticValue(descriptor.Id, out var configuredSeverity);
```

Comment:
Consider explicitly discarding to acknowledge you're deliberately ignoring this result rather than accidentally doing so. 
```suggestion
                    _ = options.TryGetGlobalDiagnosticValue(descriptor.Id, out var configuredSeverity);
```

Reply:
The way you are using `configuredSeverity` here implies that on a `false` return the value is guaranteed to be `default`. What happens if we pass a non-defaulted value into the API. Does it `default` the value?

Reply:
It explicitly sets it to `Default` (see [https://github.com/dotnet/roslyn/pull/45871/files/55ac7e5acdd2a1f91625ce31caa99af5cc43b5b9#diff-ffb0437c61e5f1029ae0c11cb59507d9R89](https://github.com/dotnet/roslyn/pull/45871/files/55ac7e5acdd2a1f91625ce31caa99af5cc43b5b9#diff-ffb0437c61e5f1029ae0c11cb59507d9R89)). It's an `out` param though, so I don't understand how we'd pass in an explicit value?



---
In reply to: [475295255](https://github.com/dotnet/roslyn/pull/45871#discussion_r475295255) [](ancestors = 475295255)